test/test_linalg.py

```diff
@@ -5936,11 +5943,22 @@ def _test_addmm_addmv(self, f, t, m, v, *, alpha=None, beta=None, transpose_out=
         else:
             alpha = 1.2 if alpha is None else alpha
             beta = 0.8 if beta is None else beta
-        res1 = f(t, m, v, alpha=alpha, beta=beta)
+
+        # float16 inputs with cusparse backend require larger tolerance than with cublas
+        if layout == torch.sparse_csr and m.dtype == torch.float16:
```

Comment:
This seems fishy. Do you know why?

Reply:
I don't know the reason other than cusparse performs multiplications and accumulation in different order.

Reply:
Maybe cc @xwang233